docker-compose.yml

```diff
@@ -8,52 +8,35 @@ services:
         CONTAINER_USER: $CONTAINER_USER
     image: oriserobotics/ros-$ROS_DISTRO:devel
     environment:
-      - NVIDIA_VISIBLE_DEVICES=all
-      - NVIDIA_DRIVER_CAPABILITIES=all
       - DISPLAY
       - COLCON_WORKSPACE_FOLDER=$COLCON_WORKSPACE_FOLDER
       - USER=$CONTAINER_USER
-      - SSH_AUTH_SOCK=$SSH_AUTH_SOCK_CONTAINER_PATH
-    network_mode: host
     volumes:
-      - $SSH_AUTH_SOCK_HOST_PATH:$SSH_AUTH_SOCK_CONTAINER_PATH
-      - ~/.gnupg:/home/$CONTAINER_USER/.gnupg:rw
-      - ~/.gitconfig:/home/$CONTAINER_USER/.gitconfig:rw
+      - home:/home/$CONTAINER_USER:rw
       - /tmp/.X11-unix:/tmp/.X11-unix:rw
-      - $VOLUMES_FOLDER/$CONTAINER_NAME:/home/$CONTAINER_USER:rw
       - /etc/localtime:/etc/localtime:ro
-    privileged: true
     working_dir: $COLCON_WORKSPACE_FOLDER
-    container_name: $CONTAINER_NAME
     stdin_open: true # docker run -i
     tty: true # docker run -t
-    deploy:
-      resources:
-        reservations:
-          devices:
-            - capabilities:
-              - gpu
+    tmpfs: /tmp
+    user: $CONTAINER_USER
+
   test:
     build:
       context: .
       dockerfile: Dockerfile.test
       args:
         ROS_DISTRO: $ROS_DISTRO
-    runtime: nvidia
     environment:
-      - NVIDIA_VISIBLE_DEVICES=all
-      - NVIDIA_DRIVER_CAPABILITIES=all
       - DISPLAY
-    network_mode: host
     volumes:
-      - ~/.ssh/known_hosts:/root/.ssh/known_hosts:rw
```

Comment:
I saw that you removed the `gosu` call, and implemented just the usage of the `ssh agent` we should be very careful with that concerning the user authorization, because we `build` and `deploy` the images with a specific `UID`, if someone does `docker pull`, and has a `UID` different of 1000, will not be able to access the `ssh_agent`, one way to workaround that is with `gosu`, or sharing keys with the needed user authorization. 

The way it is today I think only someone with the `UID=1000` will be able to use it as it is.

Reply:
in which circumstances the UID would be different than 1000 (besides running as root/sudo)

Reply:
If you are not the "first added user", can be a server used by multiple persons per example

Reply:
Many of the changes I made was to address the situation the host wants to use the tool from multiple user sessions (my case). The primary adjustment was setting the container user id with caller UID to permit the host to access the bound volume if he wants. I didn't have problems with ssh-agent having container user with id 1000 and host's 1001.

Reply:
In #57, I propose moving the user configuration (including its creation) to the entry point so that we would use `gosu` there, but I think it increases complexity.

Maybe the best is to set the user UID/GID in the entry point (then use `gosu`). We would also need to adjust the permission of the home folder there.